components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/hardware-wallet-connect/accounts-list.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react'
+import { create } from 'ethereum-blockies'
+import { Checkbox, Select } from 'brave-ui/components'
+import {
+  ButtonsContainer,
+  DisclaimerWrapper,
+  HardwareWalletAccountCircle,
+  HardwareWalletAccountListItem,
+  HardwareWalletAccountListItemColumn,
+  HardwareWalletAccountsList,
+  SelectWrapper
+} from './style'
+import {
+  HardwareWallet,
+  HardwareWalletAccount,
+  HardwareWalletDerivationPathLocaleMapping,
+  HardwareWalletDerivationPathsMapping
+} from './types'
+import { reduceAddress } from '../../../../../utils/reduce-address'
+import locale from '../../../../../constants/locale'
+import { NavButton } from '../../../../extension'
+import { SearchBar } from '../../../../shared'
+import { DisclaimerText } from '../style'
+
+interface Props {
+  hardwareWallet: HardwareWallet
+  accounts: Array<HardwareWalletAccount>
+  onLoadMore: () => void
+  selectedDerivationPaths: string[]
+  setSelectedDerivationPaths: (paths: string[]) => void
+  selectedDerivationScheme: string
+  setSelectedDerivationScheme: (scheme: string) => void
+  onAddAccounts: () => void
+}
+
+export default function (props: Props) {
+  const {
+    hardwareWallet,
+    selectedDerivationScheme,
+    setSelectedDerivationScheme,
+    setSelectedDerivationPaths,
+    selectedDerivationPaths,
+    onLoadMore,
+    onAddAccounts
+  } = props
+  const derivationPathsEnum = HardwareWalletDerivationPathsMapping[hardwareWallet]
+
+  const onSelectAccountCheckbox = (account: HardwareWalletAccount) => () => {
+    const { derivationPath } = account
+    const isSelected = selectedDerivationPaths.includes(derivationPath)
+    const updatedPaths = isSelected
+      ? selectedDerivationPaths.filter((path) => path !== derivationPath)
+      : [...selectedDerivationPaths, derivationPath]
+
+    setSelectedDerivationPaths(updatedPaths)
+  }
+
+  return (
+    <>
+      <SelectWrapper>
+        <Select value={selectedDerivationScheme} onChange={setSelectedDerivationScheme}>
+          {Object.keys(derivationPathsEnum).map((path, index) => {
+            const pathValue = derivationPathsEnum[path]
+            const pathLocale = HardwareWalletDerivationPathLocaleMapping[pathValue]
+            return (
+              <div data-value={pathValue} key={index}>
+                {pathLocale}
+              </div>
+            )
+          })}
+        </Select>
+      </SelectWrapper>
+      <DisclaimerWrapper>
+        <DisclaimerText>{locale.switchHDPathTextHardwareWallet}</DisclaimerText>
+      </DisclaimerWrapper>
+      <SearchBar placeholder={locale.searchScannedAccounts} />
```

Comment:
Can we add a `filter` method here to pass into the `action` prop for `SearchBar`

Reply:
Done! :+1: